elliottlib/cli/advisory_commons_cli.py

```diff
@@ -0,0 +1,103 @@
+              help="Select given advisory id(s). comma separated")
+@click.option("--field", required=True,
+              help="Advisory field (like publish_date) to show or update. Examples of show only fields: ["
+                   "'errata_state', "
+                   "'errata_name', 'content_types', 'synopsis', "
+                   "'errata_bugs', "
+                   "'errata_type']")
+@click.option("--new",
+              help=f"New value of field to update advisories with. --field should be one of"
+                   f" {supported_update_fields}")
+@click.option("--noop", "--dry-run",
+              is_flag=True,
+              default=False,
+              help="Do not actually change anything")
+@click.pass_obj
+def advisory_commons_cli(runtime, advisories, field, new, noop):
+    """Display or Change a common field (like date) across multiple advisories.
+
+Advisories created for an OCP version have common fields, that sometimes
+will need updating. This command helps with that.
+
+    NOTE: The two advisory input options (--assembly and --advisories)
+    are mutually exclusive and can not be used together.
+
+    Show the field "publish_date" for all advisories for an assembly
+
+    $ elliott -g openshift-4.8 --assembly 4.8.8 advisory-commons --field "publish_date"
+
+    Change the field "publish_date" for all advisories for an assembly
+
+    $ elliott -g openshift-4.8 --assembly 4.8.8 advisory-commons --field "publish_date" --new "2021-Aug-31"
+
+    Do not actually change anything
+
+    $ elliott -g openshift-4.8 --assembly 4.8.8 advisory-commons --field "publish_date" --new "2021-Aug-31" --noop
+"""
+    count_flags = sum(map(bool, [runtime.assembly != 'stream', advisories]))
+    if count_flags > 1:
+        raise click.BadParameter("Use only one of --assembly or --advisories")
+
+    if new:
+        if field not in supported_update_fields:
+            raise click.BadParameter(f"Only these fields are supported for update: {supported_update_fields}")
+    if not advisories:
+        runtime.initialize()
+        advisories = runtime.group_config.advisories.values()
+    else:
+        advisories = [a.strip() for a in advisories.split(',')]
+
+    errors = []
+    for advisory_id in advisories:
+        try:
+            e = Erratum(errata_id=advisory_id)
+            current = getattr(e, field)
+
+            if new:
+                if new == current:
+                    green_prefix(f"No Change: {advisory_id}: ")
+                    click.echo(f"Target field state is same as current state: {field}: {current}")
+                else:
+                    if noop:
+                        green_prefix(f"NOOP: {advisory_id}: ")
+                        click.echo(f"Would have changed field {field} {current} ➔ {new}")
+                    else:
+                        update_kwarg = {}
+                        if field == 'publish_date':
+                            update_kwarg['date'] = new
```

Comment:
I didn't see a compelling reason to rename this field. We have called this field `date` or `release_date` elsewhere in Elliott code.

Reply:
The weird thing is errata_tool's advisory.update() method searches for "date" specifically, any other field name is ignored.